src/hotspot/share/memory/memoryReserver.cpp

```diff
@@ -426,19 +426,17 @@ ReservedSpace HeapReserver::Instance::try_reserve_range(char *highest_start,
                                                         size_t size,
                                                         size_t alignment,
                                                         size_t page_size) {
+  assert(is_aligned(highest_start, attach_point_alignment), "precondition");
+  assert(is_aligned(lowest_start, attach_point_alignment), "precondition");
+
   const size_t attach_range = highest_start - lowest_start;
-  // Cap num_attempts at possible number.
-  // At least one is possible even for 0 sized attach range.
   const uint64_t num_attempts_possible = (attach_range / attach_point_alignment) + 1;
   const uint64_t num_attempts_to_try   = MIN2((uint64_t)HeapSearchSteps, num_attempts_possible);
+  const uint64_t num_intervals = num_attempts_to_try - 1;
```

Comment:
pre-existing, but stands out more now.  Why the introduction of uint64_t in here?  It seems like this should
just be using size_t throughout.

Reply:
Thanks, size_t if more general than uint64_t. uint64_t has been replaced as size_t.